desc/magnetic_fields/_current_potential.py

```diff
@@ -696,3 +902,649 @@ def nfp_loop(j, f):
+    normalize : bool, optional
+        whether or not to normalize Bn when printing the Bnormal errors. If true,
+        will normalize by the average equilibrium field strength on the surface.
+    vacuum : bool, optional
+        if True, will not include the contribution to the normal field from the
+        plasma currents.
+
+
+    Returns
+    -------
+    surface_current_field : FourierCurrentPotentialField or list of
+        A FourierCurrentPotentialField with the Phi_mn set to the
+        optimized current potential. If multiple alpha were passed in,
+        this is a list of length alpha.size with the optimized fields
+        for each parameter value alpha.
+    data : dict
+        Dictionary with the following keys,::
+
+            alpha : regularization parameter the algorithm was ran with, a float
+                    if passed-in alpha was a float, or an array if it was an array,
+                    corresponding to the list of `Phi_mn`.
+            Phi_mn : the single-valued current potential coefficients which
+                    minimize the Bn at the given eval_grid on the plasma, subject
+                    to regularization on the surface current magnitude governed by
+                    alpha.
+                    An array of length `self.Phi_basis.num_modes` if passed-in alpha,
+                    was a float, or a list of arrays, with list length `alpha.size` if
+                    `alpha` was an array, corresponding to the list of regularization
+                    parameters alpha.
+            I : float, net toroidal current (in Amperes) on the winding surface.
+                Governed by the `current_helicity` parameter, and is zero for
+                modular coils (`current_helicity=0`).
+            G : float, net poloidal current (in Amperes) on the winding surface.
+                Determined by the equilibrium toroidal magnetic field, as well as
+                the given external field.
+            chi^2_B : quadratic flux integrated over the plasma surface.
+                a float if `alpha` was a float, or list of float of length
+                `alpha.size` if `alpha` was an array, corresponding to the array
+                of `alpha` values.
+            chi^2_K : Current density magnitude integrated over winding surface.
+                a float if `alpha` was a float, or list of float of length
+                `alpha.size` if `alpha` was an array, corresponding to the array of
+                `alpha`.
+            |K| : Current density magnitude on winding surface, evaluated at the
+                given `source_grid`. An array of length `source_grid.num_nodes` if
+                `alpha` was a float, or list of arrays, with list length
+                `alpha.size`, if `alpha` was an array, corresponding to the array of
+                `alpha`.
+            eval_grid: Grid object that Bn was evaluated at.
+            source_grid: Grid object that Phi and K were evaluated at.
+
+
```

Comment:
should maybe include a note that this function is not compatible with jit/AD etc? (is it?)

Reply:
It is not